Tests/SwiftGRPCNIOTests/NIOClientClosedChannelTests.swift

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+import SwiftGRPCNIO
+import NIO
+import XCTest
+
+class NIOClientClosedChannelTests: NIOEchoTestCaseBase {
+  func testUnaryOnClosedConnection() throws {
+    let initialMetadataExpectation = self.makeInitialMetadataExpectation()
+    let responseExpectation = self.makeResponseExpectation()
+    let statusExpectation = self.makeStatusExpectation()
+
+    self.client.connection.close().map {
+      self.client.get(Echo_EchoRequest(text: "foo"))
+    }.whenSuccess { get in
+      get.initialMetadata.assertError(fulfill: initialMetadataExpectation)
+      get.response.assertError(fulfill: responseExpectation)
+      get.status.map { $0.code }.assertEqual(.unavailable, fulfill: statusExpectation)
+    }
+
+    self.wait(for: [initialMetadataExpectation, responseExpectation, statusExpectation],
```

Comment:
Does this work on Linux now (probably since Swift 5)?

Reply:
Yes, since Swift 5.